File: Frontend/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http:HttpClient) { }

  //get all quiz
  public getQuizzez()
  {
     return this.http.get(`${baseUrl}/quiz/`);
  }

  // add new quiz
  public addQuiz(quiz:any)
  {
      return this.http.post(`${baseUrl}/quiz/`,quiz);
  }

  //delete quiz by id.
  public deleteQuiz(qid:number)
  {
      return this.http.delete(`${baseUrl}/quiz/${qid}`);
  }
  
  //get quizzes by category
  public getQuizzesBycategory(id:any)
  {
    return this.http.get(`${baseUrl}/quiz/category/${id}`);
  }

  // get quiz by quiz-id
  public getQuizById(qid:any)
  {
    return this.http.get(`${baseUrl}/quiz/${qid}`);
  }

  
  
  

}

File: Backend/ExamPortalApplication/src/main/java/com/exam/service/QuizService.java
package com.exam.service;

import java.util.List;
import java.util.Set;

import com.exam.entity.Category;
import com.exam.entity.Question;
import com.exam.entity.Quiz;

public interface QuizService {
	
	
	public Quiz addQuiz(Quiz q);
	public void deleteQuiz(Long id);
	public Quiz getQuizById(Long id);
	public Quiz update(Quiz q);
	public Set<Quiz> getAllQuizez();
	public Set<Question> getQuestionsByQuizId(Long quizId);
	public List<Quiz> getQuizzesByCategory(Category c);
	
	
	
	
	
	
	
	

}
